test/checkFlags.test.js

```diff
@@ -0,0 +1,64 @@
+/* global describe: true, it; true */
+
+'use strict';
+
+var expect = require('expect.js'),
+    make = require('../lib/checkFlags').make,
+    compare = require('../lib/checkFlags').compare;
+
+describe('checkFlags', function () {
+
+    var DEBUG_MODE = 0x1;
+    var DIAGNOSTIC_MODE = 0x2;
+
+    var constants = {
+        FLAGS: {
+            ACTION1: make(0x1, DEBUG_MODE, 'ONE'),
+            ACTION2: make(0x2, DEBUG_MODE, 'TWO'),
+            ACTION3: make(0x4, DIAGNOSTIC_MODE, 'THREE'),
+            ACTION4: make(0x8, DIAGNOSTIC_MODE, 'FOUR'),
+            ACTION5: make(0x10, DEBUG_MODE, 'FIVE'),
+            ACTION6: make(0x20, DEBUG_MODE, 'SIX'),
+            ACTION7: make(0x40, DIAGNOSTIC_MODE, 'SEVEN'),
+            ACTION8: make(0x80, DIAGNOSTIC_MODE, 'EIGHT'),
+            ACTION9: make(0x100, DEBUG_MODE, 'NINE'),
+            ACTION0: make(0x200, DEBUG_MODE, 'TEN')
+        },
+        GLOBAL_BITMASK: 0x1,
```

Comment:
The global bitmasks for mode and flags should contain more than one option bit to check against. None of the listed events have more than one mode applied to them. Action0 should be at the top of the list, otherwise it should be Action10.